app/src/main/java/com/quran/labs/androidquran/ui/util/ToastCompat.java

```diff
@@ -0,0 +1,83 @@
+package com.quran.labs.androidquran.ui.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
+import androidx.appcompat.content.res.AppCompatResources;
+import androidx.core.content.ContextCompat;
+import androidx.core.graphics.drawable.DrawableCompat;
+
+import com.quran.labs.androidquran.R;
+import com.quran.labs.androidquran.util.QuranSettings;
+
+public class ToastCompat {
+
+  private static final int ANDROID_R = 30;
+
+  public static Toast makeText(@NonNull Context context, @StringRes int messageRes, int duration) {
+    return makeText(context, context.getString(messageRes), duration);
+  }
+
+  public static Toast makeText(@NonNull Context context, CharSequence message, int duration) {
+    @SuppressLint("ShowToast") Toast toast = Toast.makeText(context, message, duration);
+    if (Build.VERSION.SDK_INT < ANDROID_R) {
+      // It's unclear what behavior `Toast.setView()` has on R and above, hence the guard.
+      toast.setView(createToastView(context, message));
+    }
+    return toast;
+  }
+
+  private static View createToastView(Context context, CharSequence message) {
+    View toastView = LayoutInflater.from(context).inflate(R.layout.toast, null, false);
+
+    QuranSettings settings = QuranSettings.getInstance(context);
+    Drawable tintedBackground = createTintedBackground(context, settings.isNightMode());
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+      toastView.setBackground(tintedBackground);
+    } else {
+      toastView.setBackgroundDrawable(tintedBackground);
+    }
+
+    TextView textView = toastView.findViewById(android.R.id.message);
+    textView.setTextColor(getTextColor(context, settings));
+    textView.setText(message);
+    return toastView;
+  }
+
+  private static Drawable createTintedBackground(Context context, boolean nightMode) {
+    Drawable toastFrame = DrawableCompat.wrap(AppCompatResources.getDrawable(context, R.drawable.toast_frame));
+    DrawableCompat.setTint(toastFrame, getBackgroundColor(context, nightMode));
+    return toastFrame;
+  }
+
+  private static int getTextColor(Context context, QuranSettings settings) {
+    if (settings.isNightMode()) {
+      int textColorNight = ContextCompat.getColor(context, R.color.toast_text_color_night);
+      return Color.argb(
+          settings.getNightModeTextBrightness(),
+          Color.red(textColorNight),
+          Color.green(textColorNight),
+          Color.blue(textColorNight)
```

Comment:
I've seen `getNightModeTextBrightness()` used elsewhere, but with `255` hardcoded. Is that preferred?

Reply:
i think for the alpha, it's fine to hardcode to 255 here since we don't want the background to be very transparent and perhaps not very readable? as a matter of fact i am ok with adding a toast_text_color_night (until we support day/night) and just using a nice value for that instead of making it customizable

Reply:
This function is for the text color so it will only affect the contrast ratio of the text against the background by modifying the text alpha: the background of the toast will not change transparency.

I'm not sure what the requested action is on this one - are we saying to get rid of the `getNightModeTextBrightness` like this:

```java
private static int getTextColor(Context context) {
  if (settings.isNightMode()) {
    return ContextCompat.getColor(context, R.color.toast_text_color_night);
  } else {
    return ContextCompat.getColor(context, R.color.toast_text_color);
  }
}
```

or something else?

Reply:
yes exactly - handle the same way you're handling day mode where we just provide a specific background color and text color